### ISSUE TITLE
Listen to broadcasts by bulb

### ISSUE THREAD

**Issue by RobinBol** [athombv/com.lifx#6]
Let Homey listen to broadcasts by a bulb so that its state is always synced with Homey. This will fix issues #4 and #5.

Feature request is created at https://github.com/MariusRumpf/node-lifx/issues/50.

**Comment by moekdegroot** [comment]
Hi Robin, will the lifx z strips also be supported?
I saw a request about them at the same repo. 
Just received 3 of them,  they're awsome!

**Comment by RobinBol** [comment]
Not sure, you could try, perhaps they implement the same api.

**Comment by moekdegroot** [comment]
Hope so. 
So if they pull off the multizone-packet it's automatically available in homey as well?
Sorry for perhaps asking the obvious, no coding experience at all ;)

**Comment by RobinBol** [comment]
Well no, not automatically, we'd have to update our protocol layer for that. From what I just read I think our app should sort of work with Lifx Z, could you perhaps try and let me know?

**Comment by moekdegroot** [comment]
Sure! Never crossed my mind to try, duh...😵
At work at the moment but when I'm home I'll see what happens and report back.

**Comment by moekdegroot** [comment]
Hi Robin, I'm able to add the strips as lights.
Obviously I'm not able to control the zones within the strip but turning the entire strip on/off and changing the entire strip's color is working ok.
Also when I apply a preset color scheme (multiple colors) with the official app and turn the strip off the color scheme is preserved when I use homey to turn the strip on again.
The homey app is also working for complete color changes and turning it on/off.
Shall I create a new feature request for zone control?

**Comment by RobinBol** [comment]
Ok, that is a good start. Yes you can file a feature request. There is already one filed for the library we use: https://github.com/MariusRumpf/node-lifx/issues/41, so as soon as it gets implemented I can create a dedicated LIFX Z driver!

**Comment by moekdegroot** [comment]
That's awsome!
Will create one right away. 
Thanks!

**Comment by moekdegroot** [comment]
He @RobinBol 
Like I promised in issue #7 , here are a few links I found concerning getting the status of lifx lights using the cloud. 
Have no clue if it is of any value to you (me being a non-dev) but you never know. 
https://community.smartthings.com/t/release-lifx-groups/37296?page=2
https://github.com/ericvitale/ST-LIFX-Group-of-Groups/blob/master/README.md

Best, Moek

**Comment by JesperJanssen** [comment]
Any updates about fixing the sync issue for LIFX lights?

**Comment by JesperJanssen** [comment]
@RobinBol What is the progress of this enhancement. It would be awesome if Homey can listen to the broadcast of the bulbs.

**Comment by RobinBol** [comment]
Hi, sorry so far it has still been rather low on my to do list. I did find out that the bulbs do not necessarily broadcast, so we probably need to build in polling.

**Comment by pimviergever** [comment]
Yes please! This would certainly help!

**Comment by RobinBol** [comment]
So, I found some time and added some basic polling to the Lifx bulb. However, for some reason the bulb at our office is not really responsive, offline a lot, even when controlling it via the LIFX iOS app. So I hope it works better for you guys, will publish the polling functionality as a beta in the app store, if you have feedback let me know!

**Comment by moekdegroot** [comment]
Super Robin, thx!
I'll try to find some time to test this weekend.

**Comment by JesperJanssen** [comment]
Polling seems to work pretty good. It takes only a few seconds for the status of the light to be synced with Homey. I used both the LIFX android app and Google Home to turn the LIFX lights I have on/off and change the color and the brightness.

Syncing on/off status: Works perfect
Syncing brightness: Works perfect
Syncing color: Works perfect
Syncing saturation: Works perfect
Syncing white tone (the different "whites" you can choose): Works ok. The white tones are not correctly displayed in the Homey app (warm white is displayed correctly for a few seconds and then Homey decides that the color should be some kind of blue white) but I think that has more to do with the way Athom choose to display the white tones. I think this is something which can be improved in the future.

In short polling works perfect :)

Very good work @RobinBol

**Comment by moekdegroot** [comment]
Works great indeed @RobinBol !
I can concur everything @JesperJanssen mentioned above.
Thanks very much!

**Comment by pimviergever** [comment]
Any chance polling is added to the stable version?
